File: $_algotsu-kaizen/problemo-set/maximum-difference-between-increasing-elements.c
#define MAX(X, Y) (((X) > (Y)) ? (X) : (Y))

int maximumDifference(int* nums, int numsSize) {
    int result = -1;
    int r = numsSize - 1;
    int l = r - 1;
    while (l >= 0) {
        if (nums[l] < nums[r]) result = MAX(result, nums[r] - nums[l]);
        else if (nums[l] > nums[r]) r = l;
        --l;
    }

    return result;
}

File: $_algotsu-kaizen/problemo-set/maximum-difference-between-increasing-elements.rs
impl Solution {

    pub fn maximum_difference(nums: Vec<i32>) -> i32 {
        let mut result: i32 = -1;
        let mut r = nums.len() - 1;
        let mut l = r;

        while l > 0 {
            l -= 1;

            if nums[l] < nums[r] {
                result = std::cmp::max(result, nums[r] - nums[l]);
            } else if nums[l] > nums[r] {
                r = l;
            }
        }

        result
    }
}
